### ISSUE TITLE
E2E test for auth on network policy 

### ISSUE THREAD

**Issue by meyskens** [cilium/cilium#24600]
Part of https://github.com/cilium/cilium/issues/22215

An e2e test should be added to cilium-cli when an auth handler is enabled:
- it should set `auth: mtls-spiffe` and test i pods can comminucate 
- it should check if auth actually happened (see https://github.com/cilium/cilium/issues/24598)
- an always-fail auth handler should be implemented to test regressions against the datapath

Part of this work is already done by me to test things but is blocked by changing from contract to authmap and will be continued after those PRs merge.

**Comment by meyskens** [comment]
Fixed in https://github.com/cilium/cilium-cli/pull/1505
